### ISSUE TITLE
Orca doesn't announce the file name when I press ctrl + tab

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#94360]
Issue Type: <b>Bug</b>

1. Open at  least two files in the same editor group.
2. Try switch between files using ctrl+tab or ctrl+shift+tab.

Orca does not consistently advertise file names.
CC @isidorn 
VS Code version: Code - Insiders 1.44.0-insider (0dbae2083aff6ff2190c8e2157be95458915d464, 2020-04-03T05:49:11.002Z)
OS version: Linux x64 5.5.13-arch2-1

**Comment by isidorn** [comment]
@jvesouza thanks for filling.
I can reproduce this with latest VS Code insiders. And I can not reproduce with VS Code stable, thus this was introduced with the new Quick Pick work, regression assigning to March.
@jvesouza was this workign a couple of days ago, or you do not remember it working at all?

I see that we are not properly setting the aria role to the `option`, also there is one role `presentation` in the middle of html elements.
Working on a fix...

fyi @joanmarie @bpasero @chmarti

**Comment by jvesouza** [comment]
Since I started using VSCode I have never been able to use this function consistently, but I never gave it much importance because my use of VSCode was for learning and testing.
As I now started to use VSCode effectively in my work, this behavior started to become more critical.

**Comment by isidorn** [comment]
What works are quick picks with the input box.
What does not work are quick picks with no input box (like the ctrl+tab one).
@chrmarti might you have an idea if the input box does something special

@jvesouza as a workaround please use `ctrl + p`. That should work just fine. Though we are looking into a fix here.

**Comment by isidorn** [comment]
I just verified and this works on VoiceOver. So best now would be to wait and see if @joanmarie can investigate on the Orca side.

The difference between the two cases seem to be that in the case that does not work
Element A has role `listbox` and it has focus and has properly set `activedescendent` to a child
In the case which works
Element B has role `combobox` has focus and has preoprly set `activedescendent` to a child of A.

However the combination of a `listbox` with focus via `activedescendent` to a child already nicely works in the suggestWidget.
One of the differences is `aria-haspopup` which is not present in the failing case, though the failing case does not have input so I think we are good.

**Comment by joanmarie** [comment]
The answer appears to be missing events. When I press and immediately release Ctrl+Tab here's what I see when monitoring for `object:state-changed:focused` and `object:state-changed:selected` accessibility events using a listener:

```
[list box | ] from code-insiders is now focused
[entry | bar-test.py editor] from code-insiders is now focused
```

In other words, I was in a file (foo-test.py). Pressing Ctrl+Tab caused the listbox -- but no option inside the listbox -- to become focused. When I released the key, the switched-to file became focused.

I see a similar (but ever so slightly different) lack of events when I hold down Ctrl and press and release Tab repeatedly:

```
# Press Ctrl, Press and Release Tab.
[list box | ] from code-insiders is now focused

# Press and Release Tab once.
[list item | bar-test.py] from code-insiders is now selected
[list item | bar-test.py] from code-insiders is now focused

# Press and Release Tab again.
(crickets)

# Press and Release Tab again.
(crickets)

# Press and Release Tab again.
(crickets)

# Release Ctrl
[entry | bar-test.py editor] from code-insiders is now focused
```

As you can see from the above, nothing initially claims focus other than the listbox (which Orca announces). When Tab is pressed and released the next time after that, we get events as expected, and Orca announces the newly-focused item. But after that, no events means Orca doesn't know anything has changed and thus doesn't say anything.

@isidorn Does VoiceOver keep presenting the selected item/option change if you keep pressing and releasing Tab?

**Comment by isidorn** [comment]
@joanmarie thanks for looking into it. Smells like a Chrome on linux missing event bug.
We can look into a workaround, to maybe bring this closer to the other scenario when InputBox is present. I can experiment with making some changing on Monday and also @chrmarti and @bpasero might have some ideas.

Yes, VoiceOver keeps presenting the selected option change if I keep presseing and releasing tab.

**Comment by joanmarie** [comment]
If it's a Chrome on linux missing event bug, we should get it fixed in Chrome on linux. :smile: 

I can try to come up with a small stand-alone test case later on (working on a totally different thing at the moment).

**Comment by joanmarie** [comment]
@isidorn Is there any way to try this when running VSCode via `yarn web`?

**Comment by joanmarie** [comment]
@isidorn I just came up with a simple test case that I think is in the ballpark of what we are discussing. I am seeing the expected events from Chromium. Here's the test case:

```
<style>
  [role="option"][aria-selected="true"]::after { content: "✓"; }
</style>
<script>
function changeItem() {
   current = document.getElementById("list").getAttribute("aria-activedescendant");
   if (current == "item1")
       next = "item0";
   else
       next = "item1";

   document.getElementById("list").setAttribute("aria-activedescendant", next);
   document.getElementById("combo").setAttribute("aria-activedescendant", next);
   document.getElementById(current).setAttribute("aria-selected", "false");
   document.getElementById(next).setAttribute("aria-selected", "true");
}
</script>
<input id="combo" role="combobox" aria-haspopup="true" aria-autocomplete="list" aria-activedescendant="item1" onkeydown="changeItem()">
<div id="list" tabindex="0" role="listbox" aria-activedescendant="item1">
    <div role="presentation" style="position: relative; overflow: hidden;">
       <div>
          <div role="option" id="item0" aria-selected="false">bar</div>
          <div role="option" id="item1" aria-selected="true">foo</div>
       </div>
    </div>
</div>
```

When I load the above in Chromium or in Chrome or in Firefox, every time I type a key in the input I get two events a la:

```
# Typed a letter
[list item | bar] from Google Chrome is now focused
[list item | bar] from Google Chrome is now selected

# Typed a letter
[list item | foo] from Google Chrome is now focused
[list item | foo] from Google Chrome is now selected
```

What is different between my test case and what VSCode is doing?

**Comment by bpasero** [comment]
@isidorn if the fact that the input box is hidden when quick navigating, you could simply add the `isScreenReaderOptimized` check where we hide the input box to node hide it if the screen reader is enabled:

https://github.com/microsoft/vscode/blob/7c7f74b7978a375d37543b0ff4480b3abeb463f4/src/vs/base/parts/quickinput/browser/quickInput.ts#L828

**Comment by isidorn** [comment]
@bpasero this is a good idea, I just tried it out and that indeed fixes the issue.
Is it ok if I push that?

@joanmarie thanks for looking into this! The difference is that you are having an `input` and vscode does not. Will investigate more soon..

**Comment by bpasero** [comment]
Yes ok with me.

**Comment by isidorn** [comment]
I have merge a workaround for this such that we show the input box. @jvesouza it would be cool if you try vscode insiders from Tuesday and double check if this gets fixed. I verified and for me now it is good. Closing since this issue got fixed. 
For the actual issue I plan to investigate more and comment

**Comment by isidorn** [comment]
@joanmarie just tried your example and that is exactly the difference. The input box being there, if you remove the input box you will get the original situation before our workaround fix.

**Comment by joanmarie** [comment]
If I remove the input box, I don't see the problem either:
```
<body onkeydown="changeItem()">
<div id="list" tabindex="0" role="listbox" aria-activedescendant="item1">
    <div role="presentation" style="position: relative; overflow: hidden;">
       <div>
          <div role="option" id="item0" aria-selected="false">bar</div>
          <div role="option" id="item1" aria-selected="true">foo</div>
       </div>
    </div>
</div>
</body>
```

Can you give me markup that I can run directly (i.e. outside of VSCode) where I can reproduce the problem?

Mind you, I think it's great that you found a workaround, but if the bug is indeed in Chromium, we should fix it there.

**Comment by isidorn** [comment]
@joanmarie I suggest that we wait for this one to be fixed https://github.com/microsoft/vscode/issues/94404
Once that is fixed quick pick should work fine in the web, then we can see if the issue reproduces just in plain chrome.
Creating a standalone reproducable sample might be too much work without that.
Though of course you are right and we should idealy ilfe this against Chrome if it is a Chrome issue.

**Comment by jvesouza** [comment]
@isidorn  It seems to have been fixed. However, there is something I didn't understand.
When I press ctrl + tab orca announces something like the following:
Type the name of an editor to open it.

When I hear this message I release the ctrl and tab keys and the next file is opened.

I couldn't understand why the message Type the name of an editor to open it.

**Comment by isidorn** [comment]
@jvesouza you are correct, this was an oversight from our workaround.
I have created this issue to fix this https://github.com/microsoft/vscode/issues/94587
